### PR TITLE
feat: Support thanos/grafana ingress in updated apis

### PR DESCRIPTION
**Pull request by gracedo**
## What does this PR change?
Adds support for ingresses in the thanos and grafana subcharts that are running in more recent kubernetes versions in which the original `extensions/v1beta1` version has been deprecated and removed

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Thanos and Grafana ingresses can now be deployed in newer versions of Kubernetes (1.22+) using the chart

## Links to Issues or ZD tickets this PR addresses or fixes

- https://github.com/kubecost/cost-analyzer-helm-chart/issues/1262
- https://github.com/kubecost/cost-analyzer-helm-chart/issues/1349

## How was this PR tested?
Tested installing local version of the chart with the Thanos Query ingress defined in values (on k8s 1.24)

## Have you made an update to documentation?
No

### CODE REVIEW

**Comment by gracedo** [comment]
@AjayTripathy Let me know what you think about these changes, or if you still prefer that ingress support is removed entirely from the subcharts!

**Comment by gracedo** [comment]
cc @mbolt35 @dwbrown2 mind taking a look? Or is the plan to remove ingress support instead?

**Comment by AjayTripathy** [comment]
Hey @gracedo apologies for the late response, been out for thanksgiving and catching up. I'll take a look.

**Comment by Adam-Stack-PM** [comment]
@AjayTripathy, Friendly ping. See above.

File: cost-analyzer/charts/thanos/templates/query-ingress.yml
Comment (michaelmdresser) [inline cost-analyzer/charts/thanos/templates/query-ingress.yml:127]: I think this is probably a great bug fix, just want to check w/ @AjayTripathy in case it was intentional

File: cost-analyzer/charts/thanos/templates/store-ingress.yaml
Comment (michaelmdresser) [inline cost-analyzer/charts/thanos/templates/store-ingress.yaml:123]: Likewise

**Review by michaelmdresser — APPROVED** [review]
LGTM, thanks for the contribution!

File: cost-analyzer/charts/thanos/templates/query-ingress.yml
Comment (AjayTripathy) [reply cost-analyzer/charts/thanos/templates/query-ingress.yml:127]: Uhmm you can't look at the UI via the GRPC port, so I'm not sure this was intentional. Not saying this was wrong, just wondering @gracedo what was the thinking here?

File: cost-analyzer/charts/thanos/templates/store-ingress.yaml
Comment (AjayTripathy) [reply cost-analyzer/charts/thanos/templates/store-ingress.yaml:123]: Ditto.

File: cost-analyzer/charts/thanos/templates/query-ingress.yml
Comment (gracedo) [reply cost-analyzer/charts/thanos/templates/query-ingress.yml:127]: Referencing the upstream thanos helm chart, this ingress also exists https://github.com/bitnami/charts/blob/main/bitnami/thanos/templates/query/ingress-grpc.yaml likely because some use cases require being able to access the gRPC service externally (e.g. hierarchical thanos where a central querier may need to connect to various queriers running on different cluster). I imagine this ingress was originally carried over from the upstream chart when bundled into the subchart here

**Comment by Adam-Stack-PM** [comment]
@nealormsbee, It will be your call if you want to cherry-pick this PR or wait until v1.100.

**Comment by nealormsbee** [comment]
I'm not real Thanos-savvy. @michaelmdresser can you characterize the issue potentially fixed by the http -> grpc port change? If it isn't P0~P0.5 then I'd vote to punt to v1.100. Seems like it might be, though.

**Comment by michaelmdresser** [comment]
The http -> grpc thing isn't the main thing being changed here, its actually a K8s API versioning issue on more recent versions of K8s. I think the http -> grpc thing is something we want to have clarity on before merging.

It would be amazing to have this fixed in v1.99 because otherwise ingresses for Thanos and Grafana apparently don't work, but I'm not sure if its P0.

**Comment by Adam-Stack-PM** [comment]
@michaelmdresser, How risky do you think the change is?  It is Neal's call if we cherry pick but think we would all appreciate your perspective.

**Comment by AjayTripathy** [comment]
This feels low risk enough to just cherrypick in-- @nealormsbee can you do that?

**Comment by nealormsbee** [comment]
Will do! +1 from Adam -- I don't really know what I'm looking at here so leaning on y'all. If you feel it's sufficiently low-risk then let's go for it 🙌

**Pull request merged**
